packages/fleather/lib/src/widgets/editable_text_block.dart

```diff
@@ -88,57 +89,109 @@ class EditableTextBlock extends StatelessWidget {
-      return _BulletPoint(
-        style: theme.paragraph.style.copyWith(fontWeight: FontWeight.bold),
-        width: 32,
-      );
+      return _buildBulletPointForBulletList(theme, children);
     } else if (block == ParchmentAttribute.block.code) {
-      return _NumberPoint(
-        index: index,
-        count: count,
+      return _buildNumberPointsForCodeBlock(theme, children);
+    } else if (block == ParchmentAttribute.block.checkList) {
+      return _buildCheckboxForCheckList(theme, children);
+    } else {
+      return null;
+    }
+  }
+
+  List<Widget> _buildCheckboxForCheckList(
+          FleatherThemeData theme, List<LineNode> children) =>
+      children
+          .map((node) => _CheckboxPoint(
+                width: 32,
+                value: node.style.containsSame(ParchmentAttribute.checked),
+                enabled: !readOnly,
+                onChanged: (checked) => _toggle(node, checked),
+              ))
+          .toList();
+
+  List<Widget> _buildBulletPointForBulletList(
+          FleatherThemeData theme, List<Node> children) =>
+      children
+          .map((_) => _BulletPoint(
+                style:
+                    theme.paragraph.style.copyWith(fontWeight: FontWeight.bold),
+                width: 32,
+              ))
+          .toList();
+
+  List<Widget> _buildNumberPointsForCodeBlock(
+      FleatherThemeData theme, List<LineNode> children) {
+    final leadingWidgets = <Widget>[];
+    for (final _ in children) {
+      leadingWidgets.add(_NumberPoint(
+        index: leadingWidgets.length,
         style: theme.code.style
             .copyWith(color: theme.code.style.color?.withOpacity(0.4)),
         width: 32.0,
         padding: 16.0,
         withDot: false,
-      );
-    } else if (block == ParchmentAttribute.block.checkList) {
-      return _CheckboxPoint(
-        size: 14,
-        value: node.style.containsSame(ParchmentAttribute.checked),
-        enabled: !readOnly,
-        onChanged: (checked) => _toggle(node, checked),
-      );
-    } else {
-      return null;
+      ));
     }
+    return leadingWidgets;
   }
 
-  double _getIndentWidth() {
+  List<Widget> _buildNumberPointsForNumberList(
+      FleatherThemeData theme, List<LineNode> children) {
+    final leadingWidgets = <Widget>[];
+    final levels = <int>[];
+    final indexes = <int>[];
+    for (final element in children) {
+      final currentLevel =
+          element.style.get(ParchmentAttribute.indent)?.value ?? 0;
+      var currentIndex = 0;
+
+      if (leadingWidgets.isNotEmpty) {
+        if (levels.last == currentLevel) {
+          currentIndex = indexes.last + 1;
+        } else if (levels.last > currentLevel) {
+          final lastIndex =
+              levels.lastIndexWhere((element) => element == currentLevel);
+          if (lastIndex != -1) {
+            currentIndex = indexes[lastIndex] + 1;
+          }
+        }
+      }
+
+      leadingWidgets.add(_NumberPoint(
```

Comment:
Sounds reasonable that `_NumberPoint` takes the displayed index as parameter rather than 0-based index here. Don't you think?